app/services/fetch_csv_service.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'open-uri'
+require 'net/http'
+require 'byebug'
+
+module Services
+  class FetchCsvService
+    def initialize(url)
+      @url = url
+    end
+
+    def call
+      uri = URI.parse(@url)
+      response = Net::HTTP.get_response(uri)
+      raise "#{response.code} #{response.message}" unless response.is_a?(Net::HTTPOK)
```

Comment:
As an alternative one can create a custom error for the app domain and move the message formatting there.

Reply:
Yes it would be better to define a custom exception class for reusability and consistency.